projects/js-toolkit/packages/js-toolkit-core/src/transform/json/operation/addPortletProperties.ts

```diff
@@ -0,0 +1,30 @@
+/**
```

Comment:
That's one doubt I had too, because for text I use one single operation (`addLines`) for everything and we don't have transformers for CSS and .properties, for example... I didn't make them because they are complex and we don't really need them yet, because we only add (and don't modify or delete).

Regarding your quesiton, I think I prefer having multiple operations, one for each logical thing we do, but I don't know yet if that's the best and I'm not strongly convinced about it.

Rationale: the `transform` API is a powerful way to modify project files from any devtool we make. Basically we have transformers for JSON, text, and JavaScript and the idea that made me come up with this pattern was that 90% of the times what our tools do is simply modifying/mapping files (when transpiling, modifying configuration, writing new code, ...).

So, the idea is to have a set of most common operations in `js-toolkit-core` (that will be our API library) and then some more exotic operations inside the tools (f.e: `npm-bundler` has some transformation operations that are specific to its use case and are not meant to be shared with any other tool).

But now, some doubts arise: 

1) Do we really want to have a big set of operations in our API? Given that we don't use any namespacing at all (everything is exported from `index.js`) we may arrive to a point in the future where 90% of our API are transform operations without any categorization, some of them even deprecated/not used any more.
2) Should we use any namespacing/categorization to hold these transformations (i.e: put the text ones inside a `text` property of the API, the JSON ones inside `json`, and so on, for example)?
3) Should we instead move all transformations to a different place and don't make them part of our API? (probably a different npm package, but then there's the release and dependency handling problem if we take this approach)
4) Should we aim at very generic transformations or give them a proper name that specifies exactly what they do (your original question) ?

I don't think we have enough data yet to know what is best :thinking: 

My intuition tells me that having transformations with proper names is better than having too generic ones. This is because:

1) In the end, you could have a single transformation for everything, which would totally defeat the purpose of the API (reductio ad absurdum)
2) If we have a catalog of transformations, the API is much easier to use/browse, and we can know what things we touch in the configuration by simply looking at the API.

For example, if you have `addConfigurationField.ts`, `addPkgJsonScripts.ts` and `addPortletProperties.ts` you know without looking at more code that our tools deal with configurations fields, which are inside JSON files, modify npm scripts in some places, and modify portlet properties in others. 

Also, because you know what the user is trying to do when he invokes `addConfigurationField` instead of `addPkgJsonScripts`, you can impose more restrictions on (and document) the structure with TypeScript or make more validations.

Finally, when you use this API, having this type of transformations looks better, because you would read:

```javascript
transformJson(
    packageJson,
    addConfigurationField(x),
    addPkgJsonScript(y),
    addPortletProperties(z)
)
```

Which documents itself.

However, this doesn't mean that I like having an API with myriad unrelated transformations, so not sure what to do about it, yet... I would say we should probably namespace operations to avoid polluting the root namespace of `js-toolkit-core` and see case by case if they make sense as a general purpose transformation or we prefer to generalize it or hide it inside the tools...

Reply:
Addressed in https://github.com/liferay/liferay-frontend-projects/pull/578